File: Java/Bink/src/net/qiujuer/blink/box/StringSendPacket.java
package net.qiujuer.blink.box;

import net.qiujuer.blink.listener.SendListener;

import java.io.UnsupportedEncodingException;

/**
 * String send class
 */
public class StringSendPacket extends ByteSendPacket {

    public StringSendPacket(String entity) throws UnsupportedEncodingException {
        this(entity, null);
    }

    public StringSendPacket(String entity, SendListener listener) throws UnsupportedEncodingException {
        super(entity.getBytes("UTF-8"), listener);
        mType = Type.STRING;
    }
}

File: C Sharp/Blink/Blink/Box/StringSendPacket.cs
using Net.Qiujuer.Blink.Listener;
using System;
using System.Text;

namespace Net.Qiujuer.Blink.Box
{
    public class StringSendPacket : ByteSendPacket
    {
        public StringSendPacket(String entity)
            : this(entity, null)
        {

        }

        public StringSendPacket(String entity, SendListener listener)
            : base(Encoding.UTF8.GetBytes(entity), listener)
        {
            mType = PacketType.STRING;
        }
    }
}
